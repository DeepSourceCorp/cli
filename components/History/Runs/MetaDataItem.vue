<template>
  <div class="flex items-center gap-x-1.5">
    <z-icon :icon="icon" :size="size" :color="iconColor" class="flex-shrink-0" />
    <p :class="[textColor, textSize]">
      <slot>{{ label }}</slot>
    </p>
  </div>
</template>
<script lang="ts">
import { Vue, Component, Prop } from 'nuxt-property-decorator'
import { ZIcon } from '@deepsourcelabs/zeal'

@Component({
  components: {
    ZIcon
  }
})
export default class MetaDataItem extends Vue {
  @Prop({ default: '' })
  label: string

  @Prop({ required: true })
  icon: string

  @Prop({ default: 'x-small' })
  size: string

  @Prop({ default: 'vanilla-400' })
  iconColor: string

  @Prop({ default: 'text-vanilla-400' })
  textColor: string

  @Prop({ default: 'text-xxs md:text-sm' })
  textSize: string
}
</script>
