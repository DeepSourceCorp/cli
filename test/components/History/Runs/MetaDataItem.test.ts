import { render } from '@testing-library/vue'
import MetaDataItem from '~/components/History/Runs/MetaDataItem.vue'

test('renders MetaDataItem with all prop options', () => {
  const props = {
    icon: 'antipattern',
    label: 'Antipattern'
  }

  const { html } = render(MetaDataItem, {
    props,
    stubs: {
      ZIcon: true
    }
  })

  expect(html()).toMatchSnapshot(JSON.stringify(props))
})
