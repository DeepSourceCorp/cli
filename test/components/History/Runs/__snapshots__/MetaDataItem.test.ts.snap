// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders MetaDataItem with all prop options: {"icon":"antipattern","label":"Antipattern"} 1`] = `
"<div class=\\"space-x-1\\">
  <zicon-stub icon=\\"antipattern\\" size=\\"x-small\\" position=\\"\\" color=\\"vanilla-400\\" class=\\"flex-shrink-0 inline mb-px\\"></zicon-stub> <span class=\\"text-vanilla-400 text-xs md:text-sm inline break-all\\">Antipattern</span>
</div>"
`;
